client/ayon_core/tools/publisher/control.py

```diff
@@ -2389,17 +2549,57 @@ def run_action(self, plugin_id, action_id):
 
         self.emit_card_message("Action finished.")
 
+    #TODO: Test the _is_interactive method in both interactive and batch modes.
+    @property
+    def _is_interactive(self):
+        """Check for Interactive Mode
+
+        Determine if the script is running in interactive mode
+        (as opposed to batch mode) based on the host application.
+        """
+
+        host = os.getenv('AYON_HOST_NAME')
+
+        if host is None:
+            raise EnvironmentError(
+                "Host environment variable 'AYON_HOST_NAME' is not set")
+        elif host == 'maya':
+            import maya.cmds
+            return not maya.cmds.about(query=True, batch=True)
+        elif host == 'houdini':
+            import hou
+            return hou.isUIAvailable()
+        else:
+            raise ValueError(f"Unknown host environment: {host}")
+
+
     def _publish_next_process(self):
         # Validations of progress before using iterator
         # - same conditions may be inside iterator but they may be used
         #   only in specific cases (e.g. when it happens for a first time)
 
         # There are validation errors and validation is passed
         # - can't do any progree
-        if (
-            self.publish_has_validated
-            and self.publish_has_validation_errors
-        ):
+
+        # TODO: Test the _is_interactive method in both
+        #  interactive and batch modes.
+        # Determine the conditions to stop the publish process depending
+        # on whether the mode is interactive or non-interactive
+        if self._is_interactive:
+            # In interactive mode, stop for any validation
+            # errors if validation has passed
+            stop_publish_conditions = (
+                    self.publish_has_validated
+                    and self.publish_has_validation_errors
```

Comment:
Shouldn't we add blocking errors also ?
```suggestion
                    and (self.publish_has_validation_errors or self.publish_has_validation_blocking_errors)
```

Reply:
and maybe we need to use `_ignore_non_blocking_errors` here also ?

Reply:
Ignore this, Enis and I looked at it and it appears the publish_has_validation_blocking_errors is a subset of publish_has_validation_errors.
publish_has_validation_errors contains blocking and non-blocking.